### ISSUE TITLE
Ability to close Disclosure Panel

### ISSUE THREAD

**Issue by mattgreenfield** [tailwindlabs/headlessui#270]
Are there plans to add the following to the Disclosure component?
- Close on click outside
- `close` on the render prop object so we can close the disclosure from a button with it

Or should I be managing the open/closed state in my parent component?

Thanks for the help

**Comment by RobinMalfait** [comment]
Hey! Thank you for your feature suggestion!
Much appreciated! 🙏

Can you tell me more about your use case?
Disclosures are fairly simple, and functionality like `close on outside click` doesn't really make sense. For example, when you look at an FAQ section here: 
<img width="882" alt="image" src="https://user-images.githubusercontent.com/1834413/110107529-4aa2e080-7dab-11eb-9b56-b9e884336af7.png">

If you can tell me more about your use case, then maybe I can recommend a better component for you!

**Comment by mattgreenfield** [comment]
Thanks @RobinMalfait. There are actually a couple of examples in the Github UI that I'm trying to replicate.

They are panels that open and have interactive components in them. The components can close the panel but clicking outside also closes them.

<img width="509" alt="Screenshot 2021-03-05 at 11 25 18" src="https://user-images.githubusercontent.com/7290654/110109652-a7030180-7da5-11eb-9b22-04d2626bfc0a.png">
<img width="694" alt="Screenshot 2021-03-05 at 11 25 05" src="https://user-images.githubusercontent.com/7290654/110109646-a5d1d480-7da5-11eb-87ad-7a9ceccbb7b8.png">

**Comment by RobinMalfait** [comment]
Gotcha! I would recommend to use a Popover for that (https://github.com/tailwindlabs/headlessui/tree/develop/packages/@headlessui-react#popover). It is built on top of the Disclosure pattern, but it has "outside click" and "escape" to close functionality.

I'll make sure that I add a list of features for each component and when to use them to the docs!

**Comment by mattgreenfield** [comment]
Great, thanks for that.
It looks like I can use another Popover.Button inside the Popover.Panel to act as a 'close' button but what if I wan't to wait on something else before closing? e.g. a network request or form validation? 

I think there is also a bug with the Popover as I can't close it by clicking the trigger button again. It seems to close on mouse down and then open again on mouse up - possibly related to https://github.com/tailwindlabs/headlessui/issues/95
I can create a new issue with repro if you wish?
![Mar-05-2021 12-28-23](https://user-images.githubusercontent.com/7290654/110115674-547a1300-7dae-11eb-9c62-f754d02b683d.gif)

**Comment by RobinMalfait** [comment]
That should not be related no, can you provide a reproduction repo/codesandbox?

**Comment by mattgreenfield** [comment]
Thanks @RobinMalfait, here's a codesandbox- https://codesandbox.io/s/headlessuireact-menu-example-forked-41ckj
It only happens with my custom Button component, not a 'html' button - I assume it's a re-rendering issue?

**Comment by RobinMalfait** [comment]
The issue is that the ref is not forwarded to your button, here is a fix: https://codesandbox.io/s/headlessuireact-popover-component-forked-6d2bf?file=/src/App.js

I'll make sure that we start throwing errors so that this will not occur anymore, thanks!

**Comment by mattgreenfield** [comment]
Thank you!

**Comment by gilesbutler** [comment]
Hey @RobinMalfait,

Sorry to hijack an old issue but I was just wondering, in your FAQ example above is it possible to have Disclosure Panels close other panels when they're opened? I.e I click on a new FAQ and it closes the last one that I had open.

Thanks
Giles
